homeassistant/components/generic_water_heater/__init__.py

```diff
@@ -0,0 +1,54 @@
+"""The generic_water_heater integration."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.water_heater import DOMAIN as WATER_HEATER_DOMAIN
+from homeassistant.const import CONF_NAME
+from homeassistant.helpers import discovery
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = "generic_water_heater"
+
+CONF_HEATER = "heater_switch"
+CONF_SENSOR = "temperature_sensor"
+CONF_TARGET_TEMP = "target_temperature"
+CONF_TEMP_DELTA = "delta_temperature"
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                cv.slug: vol.Schema(
+                    {
+                        vol.Required(CONF_HEATER): cv.entity_id,
+                        vol.Required(CONF_SENSOR): cv.entity_id,
+                        vol.Optional(CONF_TEMP_DELTA): vol.Coerce(float),
+                        vol.Optional(CONF_TARGET_TEMP): vol.Coerce(float),
+                    }
+                )
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+async def async_setup(hass, hass_config):
```

Comment:
Migrate to `async_setup_entry`.

Reply:
Thanks for the feedback, but this integration is YAML only, so no `config_entries`